### ISSUE TITLE
vllm得到的embedding和使用transformers得到的embedding不同

### ISSUE THREAD

**Issue by Zzzhq947** [QwenLM/Qwen3-Embedding#62]
对于相同的文本列表，vllm得到的embedding和使用transformers得到的embedding不同，请问怎么修改能使其相同

**Comment by Sololololo** [comment]
same issue

**Comment by zcuder** [comment]
发现类似的问题，用Sentence  transformer算embedding跟gguf格式的ollama(llama.cpp)算出来结果不同
代码引用自https://huggingface.co/Qwen/Qwen3-Embedding-8B
`# Requires transformers>=4.51.0
# Requires sentence-transformers>=2.7.0

from sentence_transformers import SentenceTransformer

# Load the model
model = SentenceTransformer("Qwen/Qwen3-Embedding-8B")

# We recommend enabling flash_attention_2 for better acceleration and memory saving,
# together with setting `padding_side` to "left":
# model = SentenceTransformer(
#     "Qwen/Qwen3-Embedding-8B",
#     model_kwargs={"attn_implementation": "flash_attention_2", "device_map": "auto"},
#     tokenizer_kwargs={"padding_side": "left"},
# )

# The queries and documents to embed
queries = [
    "What is the capital of China?",
    "Explain gravity",
]
documents = [
    "The capital of China is Beijing.",
    "Gravity is a force that attracts two bodies towards each other. It gives weight to physical objects and is responsible for the movement of planets around the sun.",
]

# Encode the queries and documents. Note that queries benefit from using a prompt
# Here we use the prompt called "query" stored under `model.prompts`, but you can
# also pass your own prompt via the `prompt` argument
query_embeddings = model.encode(queries, prompt_name="query")
document_embeddings = model.encode(documents)

# Compute the (cosine) similarity between the query and document embeddings
similarity = model.similarity(query_embeddings, document_embeddings)
print(similarity)
# tensor([[0.7493, 0.0751],
#         [0.0880, 0.6318]])`

这个结果出来[[0.7493, 0.0751] [0.0880, 0.6318]]
如果用https://huggingface.co/Qwen/Qwen3-Embedding-8B-GGUF/tree/main 上下载的[Qwen3-Embedding-8B-Q8_0.gguf](https://huggingface.co/Qwen/Qwen3-Embedding-8B-GGUF/blob/main/Qwen3-Embedding-8B-Q8_0.gguf)
喂入ollama之后，
`
import ollama
import numpy as np
import torch
import torch.nn.functional as F  # 关键导入！

def get_detailed_instruct(task_description: str, query: str) -> str:
    return f'Instruct: {task_description}\nQuery:{query}'

# 输入数据准备（与原代码一致）
task = 'Given a web search query, retrieve relevant passages that answer the query'
queries = [
    get_detailed_instruct(task, 'What is the capital of China?'),
    get_detailed_instruct(task, 'Explain gravity')
]
documents = [
    "The capital of China is Beijing.",
    "Gravity is a force that attracts two bodies towards each other..."
]
input_texts = queries + documents

embeddings = []
for text in input_texts:
    response = ollama.embeddings(model='qwen3-embedding-8B-q5_k_m:latest', prompt=text)  # 或 model='qwen:7b'
    embeddings.append(response["embedding"])
embeddings_tensor = torch.tensor(embeddings, dtype=torch.float32)  # 显式指定数据类型[7](@ref)
embeddings_normalized = F.normalize(embeddings_tensor, p=2, dim=1)
embeddings = embeddings_normalized.numpy()  # 共享内存，修改同步[4](@ref)
scores = embeddings[:2] @ embeddings[2:].T

print(scores.tolist())
`
结果
[[0.7415439486503601, 0.2103012204170227], [0.17427761852741241, 0.467174232006073]]

我试着自己把hf模型导成gguf再导入ollama之后，同样用以上脚本计算
[[0.7617605924606323, 0.09762856364250183], [0.08815209567546844, 0.6415960788726807]]
这个结果跟sentence transformer接近，所以gguf的模型是做了什么特殊处理吗？为什么跟sentence transformer差别比较大？
@afalf

**Comment by YuXiangLin1234** [comment]
mark

**Comment by linzhengtian** [comment]
同样用qwen官方权重，我是发现用vllm（非gguf）和llama.cpp（gguf）部署的精度完全不同。
